src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -3547,13 +3552,18 @@ DeferErrorIfContainsNonPushdownableAggregate(MultiNode *logicalPlanNode)
 	MultiExtendedOp *extendedOpNode = (MultiExtendedOp *) linitial(opNodeList);
 
 	List *targetList = extendedOpNode->targetList;
+	Node *havingQual = extendedOpNode->havingQual;
 
 	/*
 	 * PVC_REJECT_PLACEHOLDERS is implicit if PVC_INCLUDE_PLACEHOLDERS isn't
 	 * specified.
 	 */
 	List *expressionList = pull_var_clause((Node *) targetList, PVC_INCLUDE_AGGREGATES |
 										   PVC_INCLUDE_WINDOWFUNCS);
+	expressionList = list_concat(expressionList,
+								 pull_var_clause(havingQual,
+												 PVC_INCLUDE_AGGREGATES |
+												 PVC_INCLUDE_WINDOWFUNCS));
```

Comment:
seems a bit odd since window functions are not allowed in HAVING even by Postgres

Reply:
Yes, will remove `INCLUDE_WINDOWFUNCS` for this call